### DISCUSSION TITLE
How to output the velocity field in compressible flow

### DISCUSSION THREAD

**Discussion by Zcaic** [su2code/SU2#1474] [Q&A]
Hello, I want to plot streamline in ParaView ,but I found no Velocity variable ，Can I output the velocity field or how plot streamline without Velocity in ParaView?

thanks!

**Comment by koodlyakshay** [comment]
You can ask for the primitive variables like velocity to be written in the paraview file by adding the keyword PRIMITIVE to the VOLUME_OUTPUT option in your config file. [See here for more details.](https://su2code.github.io/docs_v7/Custom-Output/)

**Reply by Zcaic** [reply]
thanks，But when i use 'SU2_CFD -d <config_file>' to view list of available fields, there is no **velocity**

**Reply by koodlyakshay** [reply]
You are right. I don't know why velocity is added as a primitive variable. I guess for now you might have to use the calculator on paraview to find the velocity.

**Reply by Zcaic** [reply]
OK，thanks，i will try to calculate it by myself in ParaView，I know Tecplot has this  function.

**Reply by koodlyakshay** [reply]
In Paraview you can use the 'Calculator' filter to find velocity as Momentum/Density.

**Reply by Zcaic** [reply]
Thanks

**Reply by songjuzheng** [reply]
> 
what's the unit of the velocity (Momentum/Density)? it's mach or m/s?

**Reply by pcarruscag** [reply]
For dimensional simulations it's m/s
